# MIMIC-III (<span style="color:red">ATTENTION : les données dans mimic-iii correspondent à la base de données demo, seulement 100 patients répertoriés. La base entière fait 43 Go</span>)

### Infos sur la dataset : https://physionet.org/content/mimiciii/1.4/, https://mimic.mit.edu/

26 tables... quelles sont celles qui nous intéresse ? description des tables : https://mimic.mit.edu/docs/iii/tables/

### les 2 plus importantes :
- PRESCRIPTIONS.csv : médicaments donnés à chaque patient (date de début+fin d'administration)
- DIAGNOSES_ICD.csv : maladies diagnostiquées (ICD-9 et patient id)

### à savoir : 

- PATIENTS.csv : répertorie les patients, leur genre, date d'admission+sortie
- D_ICD_DIAGNOSES.csv et D_ICD_PROCEDURES.csv : associe le code ICD9 de chaque maladie à son nom (ex:01730 <-> tuberculosis of the eye)
- DRGCODES.csv : symptomes de chaque patient (je crois)
- MICROBIOLOGYEVENTS.csv : analyses biologiques des patients 


In [1]:
import numpy as np
import pandas as pd

import matplotlib.pyplot as plt

## Exploration de DIAGNOSES_ICD.csv

In [2]:
df = pd.read_csv("mimic-iii/DIAGNOSES_ICD.csv")

In [3]:
df.head()

row_id  subject_id  hadm_id  seq_num icd9_code
0  112344       10006   142345        1     99591
1  112345       10006   142345        2     99662
2  112346       10006   142345        3      5672
3  112347       10006   142345        4     40391
4  112348       10006   142345        5     42731

In [4]:
df.shape

(1761, 5)

#### cb de maladies différentes ? 581, pour 100 patients

In [5]:
df['icd9_code'].nunique()

581

In [6]:
df['subject_id'].nunique()

100

#### -> plusieurs maladies par patient

#### repartition de ces maladies ?

In [7]:
dp = df['icd9_code'].value_counts()
dp

4019     53
42731    48
5849     45
4280     39
25000    31
         ..
8730      1
3361      1
E8231     1
34401     1
45182     1
Name: icd9_code, Length: 581, dtype: int64

In [8]:
(dp==1).value_counts()

True     322
False    259
Name: icd9_code, dtype: int64

#### -> bcp de ces 581 maladies (322) n'apparaissent qu'une fois dans la base

In [9]:
(dp<5).value_counts()

True     505
False     76
Name: icd9_code, dtype: int64

#### -> la plupart de ces maladies (505) apparaissent moins de 5 fois dans la base

In [10]:
df['subject_id'].nunique()

100

#### -> 100 patients seulement ont eu des maladies

In [14]:
patients = pd.read_csv("mimic-iii/PATIENTS.csv")

In [19]:
patients

row_id  subject_id gender                  dob                  dod  \
0     9467       10006      F  2094-03-05 00:00:00  2165-08-12 00:00:00   
1     9472       10011      F  2090-06-05 00:00:00  2126-08-28 00:00:00   
2     9474       10013      F  2038-09-03 00:00:00  2125-10-07 00:00:00   
3     9478       10017      F  2075-09-21 00:00:00  2152-09-12 00:00:00   
4     9479       10019      M  2114-06-20 00:00:00  2163-05-15 00:00:00   
..     ...         ...    ...                  ...                  ...   
95   31838       44083      M  2057-11-15 00:00:00  2114-02-20 00:00:00   
96   31853       44154      M  1878-05-14 00:00:00  2178-05-15 00:00:00   
97   31867       44212      F  2078-06-16 00:00:00  2124-01-29 00:00:00   
98   31870       44222      M  2107-06-27 00:00:00  2182-08-03 00:00:00   
99   31872       44228      F  2112-10-22 00:00:00  2171-04-14 00:00:00   

               dod_hosp              dod_ssn  expire_flag  
0   2165-08-12 00:00:00  2165-08-12 00:00:00            1  
1   2126-08-28 00:00:00                  NaN            1  
2   2125-10-07 00:00:00  2125-10-07 00:00:00            1  
3                   NaN  2152-09-12 00:00:00            1  
4   2163-05-15 00:00:00  2163-05-15 00:00:00            1  
..                  ...                  ...          ...  
95  2114-02-20 00:00:00  2114-02-20 00:00:00            1  
96  2178-05-15 00:00:00  2178-05-15 00:00:00            1  
97                  NaN  2124-01-29 00:00:00            1  
98  2182-08-03 00:00:00                  NaN            1  
99  2171-04-14 00:00:00                  NaN            1  

[100 rows x 8 columns]

# ML

https://github.com/MLforHealth/MIMIC_Extract/tree/master/notebooks : Baselines for Mortality and LOS prediction (sklearn)